### PR TITLE
Feat/37 list of routes

### PR DESCRIPTION
**Pull request by mlauren**
## Description

Draft PR for list of routes

## Original issue(s)
department-of-veterans-affairs/va-forms-system-core#37

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link to the original github issue has been provided
- [ ] No sensitive information (i.e. PII/credentials/internal URLs etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/routing/Router.tsx [hunk 1]
```diff
@@ -1,38 +1,66 @@
...
+  Routes,
+} from 'react-router-dom';
 import { Formik } from 'formik';
 import { RouterProps } from './types';
 
 import FormTitle from '../form-layout/FormTitle';
 import FormFooter from '../form-layout/FormFooter';
+import { RouterContext } from './RouterContext';
+
+const routeObjectsReducer = (routeObjectsArray: RouteObject[]) => {
```
Comment (bjmccotter7192) [inline src/routing/Router.tsx:15]: Could we add some ts docs to give a quick example of what the data object looks like? Or at least explain how what this function is returning.

File: src/routing/Router.tsx [hunk 2]
```diff
@@ -1,38 +1,66 @@
...
+  return routeObjectsArray.reduce<string[]>(
+    (accumulator, current): string[] => {
+      if (current.children) {
+        return [
```
Comment (bjmccotter7192) [inline src/routing/Router.tsx:19]: is there any way we can test the logic for this function?

**Review by bjmccotter7192 — COMMENTED** [review]
Looking good just left a couple comments.

File: src/routing/Router.tsx (hunk 1 above)
Comment (mlauren) [reply src/routing/Router.tsx:15]: yes, oops forgot to build the docs

File: src/routing/Chapter.tsx [hunk 3]
```diff
@@ -1,5 +1,11 @@
-import React from 'react';
-import { Link, Outlet } from 'react-router-dom';
+import React, { ReactElement, useContext } from 'react';
+import {
+  createRoutesFromChildren,
+  Link,
```
Comment (micahchiang) [inline src/routing/Chapter.tsx:4]: These are unused and can be removed

File: src/routing/Page.tsx [hunk 4]
```diff
@@ -1,7 +1,14 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import { useFormikContext, Form } from 'formik';
-import { useNavigate, Link, To } from 'react-router-dom';
+import {
+  useNavigate,
+  Link,
+  To,
+  useLocation,
+  RouteObject,
```
Comment (micahchiang) [inline src/routing/Page.tsx:8]: This is unused and can be removed

File: src/routing/Router.tsx (hunk 2 above)
Comment (micahchiang) [reply src/routing/Router.tsx:19]: Yeah plus 1 for adding tests around this 👍

**Review by micahchiang — APPROVED** [review]
It looks good to me, nice work @mlauren. Just left a couple comments around unused imports and adding tests for the routeObjectsReducer function

**Pull request merged**